Userland/Libraries/LibWeb/ARIA/AriaData.h

```diff
@@ -0,0 +1,273 @@
+    Vector<DeprecatedString> aria_flow_to_or_default() const;
+    Optional<bool> aria_grabbed_or_default() const;
+    AriaHasPopup aria_has_popup_or_default() const;
+    Optional<bool> aria_hidden_or_default() const;
+    AriaInvalid aria_invalid_or_default() const;
+    DeprecatedString aria_key_shortcuts_or_default() const;
+    DeprecatedString aria_label_or_default() const;
+    Vector<DeprecatedString> aria_labelled_by_or_default() const;
+    Optional<i32> aria_level_or_default() const;
+    AriaLive aria_live_or_default(Optional<AriaLive> default_value = {}) const;
+    bool aria_modal_or_default() const;
+    bool aria_multi_line_or_default() const;
+    bool aria_multi_selectable_or_default() const;
+    AriaOrientation aria_orientation_or_default(Optional<AriaOrientation> default_value = {}) const;
+    Vector<DeprecatedString> aria_owns_or_default() const;
+    DeprecatedString aria_placeholder_or_default() const;
+    Optional<i32> aria_pos_in_set_or_default() const;
+    Tristate aria_pressed_or_default() const;
+    bool aria_read_only_or_default() const;
+    Vector<AriaRelevant> aria_relevant_or_default() const;
+    bool aria_required_or_default() const;
+    DeprecatedString aria_role_description_or_default() const;
+    Optional<i32> aria_row_count_or_default() const;
+    Optional<i32> aria_row_index_or_default() const;
+    Optional<i32> aria_row_span_or_default() const;
+    Optional<bool> aria_selected_or_default() const;
+    Optional<i32> aria_set_size_or_default() const;
+    AriaSort aria_sort_or_default() const;
+    Optional<f64> aria_value_max_or_default(Optional<f64> default_value = {}) const;
+    Optional<f64> aria_value_min_or_default(Optional<f64> default_value = {}) const;
+    Optional<f64> aria_value_now_or_default() const;
+    DeprecatedString aria_value_text_or_default() const;
+
+private:
+    // https://www.w3.org/TR/wai-aria-1.2/#valuetype_true-false
+    // The default value for this value type is false unless otherwise specified.
+    static bool parse_true_false(DeprecatedString const&, bool default_value = false);
+
+    // https://www.w3.org/TR/wai-aria-1.2/#valuetype_tristate
+    // The default value for this value type is undefined unless otherwise specified.
+    static Tristate parse_tristate(DeprecatedString const&, Tristate default_value = Tristate::Undefined);
+
+    // https://www.w3.org/TR/wai-aria-1.2/#valuetype_true-false-undefined
+    // The default value for this value type is undefined unless otherwise specified.
+    static Optional<bool> parse_true_false_undefined(DeprecatedString const&, Optional<bool> default_value = {});
+
+    // https://www.w3.org/TR/wai-aria-1.2/#valuetype_integer
+    static Optional<i32> parse_integer(DeprecatedString const&);
+
+    // https://www.w3.org/TR/wai-aria-1.2/#valuetype_number
+    static Optional<f64> parse_number(DeprecatedString const&, Optional<f64> default_value = {});
+
+    DeprecatedString m_aria_active_descendant;
```

Comment:
Storing all of these fields as strings feels a bit excessive when they get converted to other types when requested. Can we just store those other types instead

I'm not super familiar with ARIA, but my understanding is that most Dom nodes will have this data attached to them, and >40 Strings is going to be very heavy.

Reply:
By [definition](https://www.w3.org/TR/wai-aria-1.2/#aria-roletype) the ARIAMixin stores all ARIA attributes as DOMStrings. Fortunately, there are a couple of ameliorating factors, first the data is specified to live as attributes in Elements, so storage is sparse. The other factor is that most elements in the real world do not have anywhere close to all ARIA properties and state specified on them.
That said, it may make sense to have AriaData hold parsed data depending on our discussion on the other thread RE is purpose.